code/datums/mapgen/Cavegens/WasteplanetCaves.dm

```diff
@@ -0,0 +1,56 @@
+/datum/map_generator/cave_generator/wasteplanet
+	open_turf_types = list(/turf/open/floor/plating/asteroid/wasteplanet = 50,
+						/turf/open/floor/plating/rust/wasteplanet = 10,
+						/turf/open/floor/plating/wasteplanet = 5)
+
+	closed_turf_types =  list(/turf/closed/mineral/random/asteroid/wasteplanet = 45,
+							/turf/closed/wall/rust = 10,)
+
+	mob_spawn_chance = 2
+	flora_spawn_chance = 5
+
+	mob_spawn_list = list(
+		//hivebots, not too difficult
+		/mob/living/simple_animal/hostile/hivebot/strong/rockplanet = 70,
+		/mob/living/simple_animal/hostile/hivebot/range/rockplanet = 40,
+		/mob/living/simple_animal/hostile/hivebot/rapid/rockplanet = 30,
+		//crystal mobs, very difficult
+		/mob/living/simple_animal/hostile/asteroid/goliath/beast/ancient/crystal = 1,
+		/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/forgotten = 1,
+		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/crystal = 1,
+		//bots, are hostile
+		/mob/living/simple_animal/bot/firebot/rockplanet = 15,
+		/mob/living/simple_animal/bot/secbot/ed209/rockplanet = 3,
+		/mob/living/simple_animal/hostile/abandoned_minebot = 15,
+		/mob/living/simple_animal/bot/floorbot/rockplanet = 15)
```

Comment:
leave a trailing comma and leave the parentheses unindented once on next line on all of these please